src/amo/components/AddonsByAuthorsCard/index.js

```diff
@@ -51,64 +63,123 @@ type Props = {|
 
 export class AddonsByAuthorsCardBase extends React.Component<Props> {
   static defaultProps = {
+    pageParam: 'page',
+    paginate: false,
+    showMore: true,
     showSummary: false,
     type: 'horizontal',
-    showMore: true,
   }
 
   componentWillMount() {
-    const { addonType, authorUsernames, forAddonSlug } = this.props;
+    const {
+      addonType,
+      authorUsernames,
+      forAddonSlug,
+      location,
+      pageParam,
+      paginate,
+    } = this.props;
 
-    this.dispatchFetchAddonsByAuthors({ addonType, authorUsernames, forAddonSlug });
+    this.dispatchFetchAddonsByAuthors({
+      addonType,
+      authorUsernames,
+      forAddonSlug,
+      page: this.getCurrentPage({ location, paginate, pageParam }),
+    });
   }
 
   componentWillReceiveProps({
     addonType: newAddonType,
     authorUsernames: newAuthorNames,
     forAddonSlug: newForAddonSlug,
+    location: newLocation,
+    pageParam,
+    paginate,
   }: Props) {
     const {
       addonType: oldAddonType,
       authorUsernames: oldAuthorNames,
       forAddonSlug: oldForAddonSlug,
+      location: oldLocation,
     } = this.props;
 
+    const newPage = paginate ?
+      oldLocation.query[pageParam] !== newLocation.query[pageParam] : false;
+
     if (
       oldAddonType !== newAddonType ||
       oldForAddonSlug !== newForAddonSlug ||
-      !deepEqual(oldAuthorNames, newAuthorNames)
+      !deepEqual(oldAuthorNames, newAuthorNames) ||
+      newPage
     ) {
       this.dispatchFetchAddonsByAuthors({
         addonType: newAddonType,
         authorUsernames: newAuthorNames,
         forAddonSlug: newForAddonSlug,
+        page: this.getCurrentPage({ location: newLocation, paginate, pageParam }),
       });
     }
   }
 
-  dispatchFetchAddonsByAuthors({ addonType, authorUsernames, forAddonSlug }: Object) {
+  getCurrentPage({
+    location, paginate, pageParam,
+  }: {|
+    location: ReactRouterLocation,
+    pageParam: string,
+    paginate: boolean,
+  |}): ?number {
+    // We don't want to set a `page` when there is no pagination.
+    if (!paginate) {
+      return undefined;
+    }
+
+    const currentPage = parseInt(location.query[pageParam], 10);
+
+    return Number.isNaN(currentPage) || currentPage < 1 ? 1 : currentPage;
+  }
+
+  dispatchFetchAddonsByAuthors({
+    addonType,
+    authorUsernames,
+    forAddonSlug,
+    page,
+  }: Object) {
```

Comment:
Can you add types for these? It may be helpful to use `$PropertyType` on the existing signature type for `fetchAddonsByAuthors`.